ipc/provider/src/checkpoint.rs

```diff
@@ -176,13 +218,20 @@ impl<T: BottomUpCheckpointRelayer + Send + Sync + 'static> BottomUpCheckpointMan
                             "cannot submit bottom up checkpoint at height {} due to: {e:}",
                             event.height
                         )
-                    })?;
-
-                log::info!(
-                    "submitted bottom up checkpoint({}) in parent at height {}",
-                    event.height,
-                    epoch
-                );
+                    }) {
+                    Ok(epoch) => {
+                        self.metrics.checkpoint_submit_succeed_count.inc();
+                        log::info!(
+                            "submitted bottom up checkpoint({}) in parent at height {}",
+                            event.height,
+                            epoch
+                        );
+                    }
+                    Err(err) => {
+                        self.metrics.checkpoint_submit_fail_count.inc();
+                        return Err(err);
+                    }
+                }
```

Comment:
We'll want to use the best practice for binary outcomes (a total count and an error count): https://promlabs.com/blog/2023/09/19/errors-successes-totals-which-metrics-should-i-expose-to-prometheus/#recommended-for-binary-outcomes-exposing-errors-and-totals

Reply:
Thanks. Fixed.